pkg/kv/kvserver/concurrency/lock_table.go

```diff
@@ -4384,3 +4385,17 @@ func MarkSkipLockedReplayError(cause error) error {
 	}
 	return errors.Mark(cause, &SkipLockedReplayError{})
 }
+
+// maxLockStrength determines the highest lock strength of amont
+// the lock holders.
+func (kl *keyLocks) maxLockStrength() (lock.Strength) {
+	str := lock.Shared
+	for e := kl.holders.Front(); e != nil; e = e.Next() {
```

Comment:
If a lock has multiple holders (e.g. shared locks), we'll want to represent that in the `LockStateInfo`. To do that, we'll need to reorganize `LockStateInfo` to contain multiple holders. We can add a new field to the message which is a slice of `lock.Holder`, where `lock.Holder` is a new proto that contains `enginepb.TxnMeta`, `lock.Durability`, `lock.Strength`, and `time.Duration`.

For backward compatibility, we can continue to populate the `LockHolder`, `Durability`, and `HoldDuration` fields with the first holder.

Making these changes will address the "This doesn't work with multiple lock holders" comment up above.